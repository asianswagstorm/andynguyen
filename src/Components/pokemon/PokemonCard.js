import React, { Component } from "react";
import "./styles/PokemonCard.css";
import spinner from "./spinner.gif";
import {pokemonImage,pokemonGif} from "./apiServices/pokeAPI";

export class PokemonCard extends Component {
  constructor(props){
    super(props);
    this.state = {
      imageLoading: true,
      gifNotFound: false,
      toManyRequests: false
    };
  };
  render() {
    const {  pokemonIndex, name } = this.props;
  
    return (
    
      <div className="individual__card">
        <a className="link-hover" href={`/Pokemon/${pokemonIndex}`}>
            <div className="card">
              <h5 className="card-header">{pokemonIndex}</h5>
              <div className="card-body">
                <div className="pokemon__image">
                  {this.state.imageLoading && (
                  <img
                    src={spinner}
                    alt={"loading gif"}
                    id="pokemonGif"
                    className="card-img-top rounded mx-auto d-block mt-2"
                  />
                )}
                {this.state.gifNotFound ? (
                  <img
                  src={pokemonImage(pokemonIndex)} 
                  alt="some pokemon"
                  id="pokemonImg"
                  onLoad={() => this.setState({ imageLoading: false })}
                  onError={() => this.setState({ toManyRequests: true })}
                />
                )  :  <img
                src={pokemonGif(name)} 
                alt="some pokemon"
                id="pokemonGif"
                onLoad={() => this.setState({ imageLoading: false })}
                onError={() => this.setState({ gifNotFound: true })}
              />}

              {this.state.toManyRequests && (
                <h6 className="mx-auto">
                  <span className="badge badge-danger mt-2">
                    Too Many Requests
                  </span>
                </h6>
              ) }

              </div>
                <h3 className="card-title"> {name.charAt(0).toUpperCase() + name.slice(1)} </h3>
              </div>
          </div>
        </a>
      </div> 
    );
  }
};

export default PokemonCard;
