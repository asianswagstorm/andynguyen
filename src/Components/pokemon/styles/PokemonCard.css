div.card,div.card-body.mx-auto{
    background-color: white;
}

.pokemon-data{
    margin: 25px 70px 70px 70px;
}

.individual__card{
    opacity: 0.95;
    transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1); 
    align-self: center;
    -moz-user-select: none; /* Mozilla-specific values */
    -ms-user-select: none; /* Microsoft-specific values */
    -webkit-user-select: none; /* WebKit-specific values */
    user-select: none; /* Keyword values */
    -o-user-select: none; 
}

div.pokemon__image{
    display: block;
    width: 100%;
    height: 200px; 
}

h5.card-header{
    width:100%;
}

img#pokemonImg{
    height: 100%;
    width:100%;
}

img#pokemonGif{
    height: 60%;
    width:60%;
    margin-left: 50px;
    margin-top: 50px;
}

h3.card-title{
    text-align: center;
    text-shadow: 2px 2px 2px darkgrey;
    color:black;
}

@media only screen and (width: 768px) and (height:1024px) {
    h3{ 
        font-size: 15px;
    }
}

.link-hover :hover, .link-hover:focus , .link-hover:visited ,.link-hover:link ,.link-hover:active  {
    box-shadow: 0 14px 28px rgba(255,255,255, 0.9), 0 10px 10px rgba(255,255,255, 0.9);
    color:black;
    text-decoration: none;
    
}

