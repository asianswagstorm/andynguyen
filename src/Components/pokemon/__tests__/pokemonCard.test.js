import React from 'react';
import {shallow,mount} from 'enzyme';
import {PokemonCard} from '../PokemonCard';
//contextType (Router!!! )
//set defaultState!!!! test different conditions

const pokemonCardComponent = <PokemonCard  name={"aPokemon"}
                                           image = {"pokemonImageLink"}
                                           pokemonIndex={1}/>

const mountPokemonCard = () => mount(pokemonCardComponent);

const shallowRenderPokemonCard = () => shallow(pokemonCardComponent);

describe('Pokemons Card', () => {
    it('should render', () => {
        const wrapper = shallowRenderPokemonCard()
        expect(wrapper).toMatchSnapshot()
    })

    it('should contain individual__card', () =>{
        const wrapper = mountPokemonCard()
        expect(wrapper.find('div.individual__card').length).toEqual(1)
    })
})

