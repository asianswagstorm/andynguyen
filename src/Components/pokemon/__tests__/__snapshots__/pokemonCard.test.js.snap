// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemons Card should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PokemonCard
    image="pokemonImageLink"
    name="aPokemon"
    pokemonIndex={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        className="link-hover"
        href="#/Pokemon/1"
>
        <div
                className="card"
        >
                <h5
                        className="card-header"
                >
                        1
                </h5>
                <div
                        className="card-body mx-auto"
                >
                        <div
                                className="pokemon__image"
                        >
                                <img
                                        alt="loading gif"
                                        className="card-img-top rounded mx-auto d-block mt-2"
                                        id="pokemonGif"
                                        src="spinner.gif"
                                />
                                <img
                                        alt="some pokemon"
                                        id="pokemonGif"
                                        onError={[Function]}
                                        onLoad={[Function]}
                                        src="https://projectpokemon.org/images/normal-sprite/aPokemon.gif"
                                />
                        </div>
                        <h3
                                className="card-title"
                        >
                                 
                                APokemon
                                 
                        </h3>
                </div>
        </div>
</a>,
      "className": "individual__card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card"
>
          <h5
                    className="card-header"
          >
                    1
          </h5>
          <div
                    className="card-body mx-auto"
          >
                    <div
                              className="pokemon__image"
                    >
                              <img
                                        alt="loading gif"
                                        className="card-img-top rounded mx-auto d-block mt-2"
                                        id="pokemonGif"
                                        src="spinner.gif"
                              />
                              <img
                                        alt="some pokemon"
                                        id="pokemonGif"
                                        onError={[Function]}
                                        onLoad={[Function]}
                                        src="https://projectpokemon.org/images/normal-sprite/aPokemon.gif"
                              />
                    </div>
                    <h3
                              className="card-title"
                    >
                               
                              APokemon
                               
                    </h3>
          </div>
</div>,
        "className": "link-hover",
        "href": "/Pokemon/1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h5
              className="card-header"
>
              1
</h5>,
            <div
              className="card-body mx-auto"
>
              <div
                            className="pokemon__image"
              >
                            <img
                                          alt="loading gif"
                                          className="card-img-top rounded mx-auto d-block mt-2"
                                          id="pokemonGif"
                                          src="spinner.gif"
                            />
                            <img
                                          alt="some pokemon"
                                          id="pokemonGif"
                                          onError={[Function]}
                                          onLoad={[Function]}
                                          src="https://projectpokemon.org/images/normal-sprite/aPokemon.gif"
                            />
              </div>
              <h3
                            className="card-title"
              >
                             
                            APokemon
                             
              </h3>
</div>,
          ],
          "className": "card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 1,
              "className": "card-header",
            },
            "ref": null,
            "rendered": 1,
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="pokemon__image"
>
                  <img
                                    alt="loading gif"
                                    className="card-img-top rounded mx-auto d-block mt-2"
                                    id="pokemonGif"
                                    src="spinner.gif"
                  />
                  <img
                                    alt="some pokemon"
                                    id="pokemonGif"
                                    onError={[Function]}
                                    onLoad={[Function]}
                                    src="https://projectpokemon.org/images/normal-sprite/aPokemon.gif"
                  />
</div>,
                <h3
                  className="card-title"
>
                   
                  APokemon
                   
</h3>,
              ],
              "className": "card-body mx-auto",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <img
                      alt="loading gif"
                      className="card-img-top rounded mx-auto d-block mt-2"
                      id="pokemonGif"
                      src="spinner.gif"
/>,
                    <img
                      alt="some pokemon"
                      id="pokemonGif"
                      onError={[Function]}
                      onLoad={[Function]}
                      src="https://projectpokemon.org/images/normal-sprite/aPokemon.gif"
/>,
                    false,
                  ],
                  "className": "pokemon__image",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "loading gif",
                      "className": "card-img-top rounded mx-auto d-block mt-2",
                      "id": "pokemonGif",
                      "src": "spinner.gif",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "some pokemon",
                      "id": "pokemonGif",
                      "onError": [Function],
                      "onLoad": [Function],
                      "src": "https://projectpokemon.org/images/normal-sprite/aPokemon.gif",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  false,
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "APokemon",
                    " ",
                  ],
                  "className": "card-title",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "APokemon",
                  " ",
                ],
                "type": "h3",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "a",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          className="link-hover"
          href="#/Pokemon/1"
>
          <div
                    className="card"
          >
                    <h5
                              className="card-header"
                    >
                              1
                    </h5>
                    <div
                              className="card-body mx-auto"
                    >
                              <div
                                        className="pokemon__image"
                              >
                                        <img
                                                  alt="loading gif"
                                                  className="card-img-top rounded mx-auto d-block mt-2"
                                                  id="pokemonGif"
                                                  src="spinner.gif"
                                        />
                                        <img
                                                  alt="some pokemon"
                                                  id="pokemonGif"
                                                  onError={[Function]}
                                                  onLoad={[Function]}
                                                  src="https://projectpokemon.org/images/normal-sprite/aPokemon.gif"
                                        />
                              </div>
                              <h3
                                        className="card-title"
                              >
                                         
                                        APokemon
                                         
                              </h3>
                    </div>
          </div>
</a>,
        "className": "individual__card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="card"
>
            <h5
                        className="card-header"
            >
                        1
            </h5>
            <div
                        className="card-body mx-auto"
            >
                        <div
                                    className="pokemon__image"
                        >
                                    <img
                                                alt="loading gif"
                                                className="card-img-top rounded mx-auto d-block mt-2"
                                                id="pokemonGif"
                                                src="spinner.gif"
                                    />
                                    <img
                                                alt="some pokemon"
                                                id="pokemonGif"
                                                onError={[Function]}
                                                onLoad={[Function]}
                                                src="https://projectpokemon.org/images/normal-sprite/aPokemon.gif"
                                    />
                        </div>
                        <h3
                                    className="card-title"
                        >
                                     
                                    APokemon
                                     
                        </h3>
            </div>
</div>,
          "className": "link-hover",
          "href": "/Pokemon/1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h5
                className="card-header"
>
                1
</h5>,
              <div
                className="card-body mx-auto"
>
                <div
                                className="pokemon__image"
                >
                                <img
                                                alt="loading gif"
                                                className="card-img-top rounded mx-auto d-block mt-2"
                                                id="pokemonGif"
                                                src="spinner.gif"
                                />
                                <img
                                                alt="some pokemon"
                                                id="pokemonGif"
                                                onError={[Function]}
                                                onLoad={[Function]}
                                                src="https://projectpokemon.org/images/normal-sprite/aPokemon.gif"
                                />
                </div>
                <h3
                                className="card-title"
                >
                                 
                                APokemon
                                 
                </h3>
</div>,
            ],
            "className": "card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 1,
                "className": "card-header",
              },
              "ref": null,
              "rendered": 1,
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="pokemon__image"
>
                    <img
                                        alt="loading gif"
                                        className="card-img-top rounded mx-auto d-block mt-2"
                                        id="pokemonGif"
                                        src="spinner.gif"
                    />
                    <img
                                        alt="some pokemon"
                                        id="pokemonGif"
                                        onError={[Function]}
                                        onLoad={[Function]}
                                        src="https://projectpokemon.org/images/normal-sprite/aPokemon.gif"
                    />
</div>,
                  <h3
                    className="card-title"
>
                     
                    APokemon
                     
</h3>,
                ],
                "className": "card-body mx-auto",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <img
                        alt="loading gif"
                        className="card-img-top rounded mx-auto d-block mt-2"
                        id="pokemonGif"
                        src="spinner.gif"
/>,
                      <img
                        alt="some pokemon"
                        id="pokemonGif"
                        onError={[Function]}
                        onLoad={[Function]}
                        src="https://projectpokemon.org/images/normal-sprite/aPokemon.gif"
/>,
                      false,
                    ],
                    "className": "pokemon__image",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "alt": "loading gif",
                        "className": "card-img-top rounded mx-auto d-block mt-2",
                        "id": "pokemonGif",
                        "src": "spinner.gif",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "alt": "some pokemon",
                        "id": "pokemonGif",
                        "onError": [Function],
                        "onLoad": [Function],
                        "src": "https://projectpokemon.org/images/normal-sprite/aPokemon.gif",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                    false,
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "APokemon",
                      " ",
                    ],
                    "className": "card-title",
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "APokemon",
                    " ",
                  ],
                  "type": "h3",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "a",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifeCycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
